pandas/core/common.py

```diff
@@ -2246,6 +2254,42 @@ def _join_unicode(lines, sep=''):
                          for x in lines])
 
 
+def _justify(text, max_len, mode='right', strlen=len):
+    """
+    Perform ljust, center, rjust against string or list-like
+    Text width is calcurated using strlen func
+    """
+
+    if not is_list_like(text):
+        # only accept list as input to avoid repeated check of padding
+        raise ValueError('input must be list-like')
+
+    # format._strlen_func return strlen wrapping with functools.partial
+    if isinstance(strlen, partial):
+        name = strlen.func.__name__
+    else:
+        name = strlen.__name__
+
+    if name == 'east_asian_len':
```

Comment:
pass a different pad function if you are in ``east_asian_len`` mode, then then you don't need the if/then block that duplicates code